addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2732,6 +2732,64 @@ QUnit.module('relational_fields', {
         form.destroy();
     });
 
+    QUnit.test('x2many default_order multiple fields', function (assert) {
+        assert.expect(18);
+
+        this.data.partner.records = [
+            {int_field: 10, id: 1, display_name: "record_form"},
+            {int_field: 10, id: 2, display_name: "record2"},
+            {int_field: 10, id: 3, display_name: "record3"},
+            {int_field: 11, id: 4, display_name: "record4"},
+            {int_field: 11, id: 5, display_name: "record5"},
+            {int_field: 11, id: 6, display_name: "record6"},
+            {int_field: 12, id: 7, display_name: "record7"},
+        ];
+
+        this.data.partner.records[0].p = _.shuffle([2, 3, 4, 5, 6, 7]);
```

Comment:
i don't like this as you don't have the control on the produced list -> i'd prefer to hardcode a specific order which reproduces the issue